multi-tenancy/README.md

```diff
@@ -0,0 +1,134 @@
+
+Our idea to handle multi-tenancy on the read path is to map group of people with tenants.
+We can retrieve those groups of people through the OAuth token received from Grafana.
+Below an example of OAuth token content coming from Dex:
+
+```json
+{
+  "iss": "https://dex.golem.gaws.gigantic.io",
+  "sub": "XXXXXXX",
+  "aud": "YYYYYYY",
+  "exp": 1705661375,
+  "iat": 1705659575,
+  "email": "people@giantswarm.io",
+  "email_verified": true,
+  "groups": [
+    "giantswarm-ad:giantswarm-admins",
+    "giantswarm-ad:GS Support - MS teams",
+    "giantswarm-ad:Giant Swarm Global",
+    "giantswarm-ad:GiantSwarm",
+    "giantswarm-ad:Giant Swarm EU",
+    "giantswarm-ad:Developers"
+  ],
+  "name": "People People"
+}
+```
+
+The component `loki-multi-tenant-proxy` should be renamed `grafana-multi-tenant-proxy` to be more generic because it will handle multi-tenancy for logs, metrics, traces, etc.
+That component should have a configuration in which the mapping between groups of people and tenants is defined.
+
+Below a proposal such of configuration:
+
+```yaml
+groups:
+- name: giantswarm-admins
+  tenant_ids:
+  - mc-name
+  - wcdev-name
+  - wcprod-name
+- name: Developers
+  tenant_ids:
+  - wcdev-name
+- name: Ops
+  tenant_ids:
+  - wcdev-name
+  - wcprod-name
+...
+```
+
+That configuration should be dynamically created and updated regardless of the data source.
+We are thinking of having an operator to manage that mapping. The operator will be able to reconcile any kind of data source and generate the configuration expected by `grafana-multi-tenant-proxy`.
+
+Below is a graph exposing multi-tenancy proposal:
+
+<img src="./assets/future-multi-tenancy.png" width="300" alt="Multi-tenancy proposal on the read path">
+
+### Opened questions
+
+#### Teleport issue
+
+We are currently facing an issue with `teleport` authentication:
+[Teleport JWT issue](https://github.com/giantswarm/giantswarm/issues/29719)
+
+We are authentified on Grafana through a JWT token provided by teleport.
+But that token is not forwarded to the datasource.
+So we are not able to authenticate the user into the multi-tenant proxy.
+We are considering a number of options that are not ideal at the moment:
+
+- having double datasource: one for customer with OAuth authentication and another dedicated to GiantSwarm people authenticated throw teleport.
+- having double grafana (same idea than above but for the all grafana instance, not just the datasource).
+- other ideas are welcome!
+
+#### Operator
+
+`logging-operator` is an existing component responsible to create a secret for the multi-tenant-proxy. That secret contains a mapping between credentials and tenant (now, it's equivalent to the cluster id).
+Do we add a feature in that operator to handle the multi-tenant-proxy configuration we described above ?
+In that case, we should rename it: any suggestion ?
+If we don't reuse `logging-operator`, we will need to implement a new one: any suggestion for its name ?
```

Comment:
This list of question in a single paragraph is a bit dense.
Maybe this should be a list?